### ISSUE TITLE
Tilemap inadequately loads CPU and GPU

### ISSUE THREAD

**Issue by NipponSunrise** [openfl/openfl#2411]
Adding openfl.display.Tilemap on stage (no matter which level) inadequately loads CPU and GPU in **any** browser.

**To Reproduce**
Run the code and see the CPU& GPU load:

	class Main extends openfl.display.Sprite {
		public function new() {
			super();

			graphics.beginFill(0xff0000);
			graphics.drawRect(0, 0, 200, 200);
			graphics.endFill();

			var tile = new openfl.display.Tilemap(1, 1);

			addEventListener(openfl.events.MouseEvent.CLICK, (e) -> {
				if (tile.parent == null) {
					addChild(tile);
					trace("Tilemap added");
				}
				else {
					removeChild(tile);
					trace("Tilemap removed");
				}
			});
		}
	}

**Sample project can be downloaded [here](https://github.com/openfl/openfl/files/5295835/Project.zip)**
Command used to build the project `haxelib run openfl build html5 -debug`

**Screenshots**

**CPU&GPU load with Tilemap on the stage:**
![image](https://user-images.githubusercontent.com/941814/94501351-4bd34e80-020a-11eb-8eda-b0390e959a3b.png)

**CPU&GPU load without Tilemap**
![image](https://user-images.githubusercontent.com/941814/94501490-9ce34280-020a-11eb-9be3-58330975d4a4.png)

**OpenFL Versions**
8.9.7 & 9.0.2

**OpenFL Targets**
html5

**Comment by dimensionscape** [comment]
There is some baseline activity involved in the renderer that needs to be executed regardless of if you're using a single tilemap or 10 tilemaps. Whether or not the cpu/gpu usage is proportional to the task remains to be seen, but I'm definitely looking into it. Unfortunately, its a little more tricky to profile openfl html5 target. If its possible to improve the baseline rendering performance, we should look into it for sure.

**Comment by dimensionscape** [comment]
Can you tell me your cpu and gpu specs so I know I'm comparing apples to apples?

**Comment by NipponSunrise** [comment]
> 
> 
> Can you tell me your cpu and gpu specs so I know I'm comparing apples to apples?

**CPU:** Core i7-8550U with 4 physical cores
**GPU:** Intel UHD 620
**RAM:** 16Gb
**Screen resolution:** 1920x1200

**Comment by NipponSunrise** [comment]
> 
> 
> There is some baseline activity involved in the renderer that needs to be executed regardless of if you're using a single tilemap or 10 tilemaps. Whether or not the cpu/gpu usage is proportional to the task remains to be seen, but I'm definitely looking into it. 

Please, note that I put on the stage **1x1 empty** Tilemap. And there is no any other activity or any actions.

Also, I think it is worth to mention my topic in the community here. Please, read it.
https://community.openfl.org/t/copypixels-movieclip-play-tilemap-load-cpu-gpu-to-much-in-html5-target/12711 

Thank you.

**Comment by dimensionscape** [comment]
> 
> 
> > There is some baseline activity involved in the renderer that needs to be executed regardless of if you're using a single tilemap or 10 tilemaps. Whether or not the cpu/gpu usage is proportional to the task remains to be seen, but I'm definitely looking into it.
> 
> Please, note that I put on the stage **1x1 empty** Tilemap. And there is no any other activity or any actions.
> 
> Also, I think it is worth to mention my topic in the community here. Please, read it.
> https://community.openfl.org/t/copypixels-movieclip-play-tilemap-load-cpu-gpu-to-much-in-html5-target/12711
> 
> Thank you.

Yes, I downloaded your example. So after profiling, I noticed that most of the performance loss is in the compositor thread. I think this is due to how the browser layers content. I don't see any openfl script taking a significant amount of frame time when the tilemap is added so I don't think this is actually an issue specific to openfl. In any case, the size of the tilemap is pretty irrelevant here.

**Comment by NipponSunrise** [comment]
> Yes, I downloaded your example. So after profiling, I noticed that most of the performance loss is in the compositor thread. I think this is due to how the browser layers content. I don't see any openfl script taking a significant amount of frame time when the tilemap is added so I don't think this is actually an issue specific to openfl. In any case, the size of the tilemap is pretty irrelevant here.

I did a profiling too. Please, take a look at my screenshot. 
![image](https://user-images.githubusercontent.com/941814/94560649-9fc24f80-026b-11eb-9b23-b6807a55d394.png)

When the Tilemap (even empty one) is on the stage canvas begins to produce 60fps for nothing. May be some enter_frame routine is needed to be optimized? Is it so necessary to redraw nothing?

**Comment by NipponSunrise** [comment]
BTW, pure `window.requestAnimationFrame` loads cpu&gpu twice less.

**Comment by dimensionscape** [comment]
https://github.com/openfl/openfl/commit/26a86a17f70b4cc34286ae25b1e6a43f67485194 Joshua seems to have made a fix that should prevent an empty tilemap from consuming resources

**Comment by loudoweb** [comment]
@NipponSunrise hi, could you check the fix please?
